### DISCUSSION TITLE
Whats the benefit of using LittleKt over LibKTX/KorGE?

### DISCUSSION THREAD

**Discussion by Its-Kenta** [littlektframework/littlekt#148] [Q&A] [answered]
What is the selling point of using LittleKt over other frameworks out there? e.g. KorGE which shares similar philosophy or the battle tested LibGDX(LibKTX)?

**Comment by LeHaine** [comment, marked as answer]
I would say the selling point on using it over Korge and LibGDX is if you are looking for a multiplatform Kotlin based game framework that is lightweight and low level enough to be able to build your own engine / framework on top of it without being stuck doing it a certain way but high level enough to not have to deal with all the low level graphics related coding. Along with getting the full benefits of the Kotlin language.

Korge is great but too high level and not as flexible to be able to build it the way you want. But for a more concrete game engine then it's great to use.

LibGDX is great that allows you to build it the way you want but doesn't allow you to target the web when using Kotlin along with it all being built in Java. I know LibKTX helps with the Java problem but doesn't help with the web issue. If those aren't an issue then libGDX would probably be a good choice.


As an example, I built my own framework on top of LittleKt which makes it a little more concrete on how to build a game with it. This is my personal way I wanted to build a game but not the way it has to be done. https://github.com/LeHaine/rune-kt

**Reply by Its-Kenta** [reply]
> I would say the selling point on using it over Korge and LibGDX is if you are looking for a multiplatform Kotlin based game framework that is lightweight and low level enough to be able to build your own engine / framework on top of it without being stuck doing it a certain way but high level enough to not have to deal with all the low level graphics related coding. Along with getting the full benefits of the Kotlin language.
> 
> Korge is great but too high level and not as flexible to be able to build it the way you want. But for a more concrete game engine then it's great to use.
> 
> LibGDX is great that allows you to build it the way you want but doesn't allow you to target the web when using Kotlin along with it all being built in Java. I know LibKTX helps with the Java problem but doesn't help with the web issue. If those aren't an issue then libGDX would probably be a good choice.

Would LittleKt still be a suitable choice for making games without having to build your own framework on top of it without much hassle? I see that there is no KMM native output with LittleKt. Is that in plans for future? Of course it using LWJGL it would prove to be impossible I suppose?

Mainly asking regarding KMM as (while it is quite awful right now) it would be useful to target certain platform where JVM is not available.

**Reply by LeHaine** [reply]
I believe it's a suitable choice. It's very similar to building a game in libGDX. I believe it offers enough tools and utilities and high enough level API to create _something_.

As for other targets excluding JVM, there are long term plans to add iOS and Kotlin/Native builds to LittleKt. Because LittleKt is still young, the focus has been trying to get the functionality in place and tightened up versus adding those extra targets. Currently, it targets JVM, Android, and Web. The rest shall come 😃.

**Reply by Its-Kenta** [reply]
Sounds great! I'm looking forward to seeing what you build and going to love trying our LittleKt. Frankly speaking KorGE caused me a tons of issues and bugs while LibGDX didn't feel very intuitive when using it with LibKTX and regardless of how much I tried, it always ended up almost looking like Java code. Or perhaps that's just me still being fresh out of the bakery.
